probe/endpoint/reporter.go

```diff
@@ -193,9 +195,11 @@ func (r *Reporter) makeEndpointNode(namespaceID string, addr string, port uint16
 	node := report.MakeNodeWith(
 		report.MakeEndpointNodeID(r.hostID, namespaceID, addr, portStr),
 		map[string]string{Addr: addr, Port: portStr})
-	// In case we have a reverse resolution for the IP, we can use it for
-	// the name...
-	if names, err := r.reverseResolver.get(addr); err == nil {
+	names := r.dnsSnooper.CachedNamesForIP(addr)
+	if resolvedNames, err := r.reverseResolver.get(addr); err == nil {
+		names = append(names, resolvedNames...)
+	}
+	if len(names) > 0 {
```

Comment:
Stuffing the names we obtained from snooping forward resolutions into the `ReverseDNSNames` set is rather misleading.

So I reckon we should either rename it (but the app will also have to pay attention to the old name, for backward compatibility), or introduce a new set, just for the snooped names.